kpi/tests/test_assets.py

```diff
@@ -779,3 +781,75 @@ def test_anonymous_as_baseline_for_authenticated(self):
         for user_obj in AnonymousUser(), self.anotheruser:
             self.assertTrue(user_obj.has_perm(
                 PERM_VIEW_ASSET, self.asset))
+
+
+class TestAssetNameSettingHandling(AssetsTestCase):
+    """
+    Tests for the 'name' setting in the asset content
+    """
+    def test_asset_name_matches_instance_root(self):
+        """
+        Test if 'name' setting is provided, it should match with the root node.
+        """
+        content = {
+            'survey': [
+                {
+                    'text': 'select_one',
+                    'label': 'q1'
+                },
+            ],
```

Comment:
Interesting! I'm surprised the unit tests don't fail since there's no `type`. I'll look into it.

Reply:
It turns out that the entire question gets silently discarded (!)
```
(Pdb) l
809             )
810             asset.deploy(backend='mock', active=True)
811             breakpoint()
812  
813             # Get the deployed XForm and parse it
814  ->         xform = XForm.objects.get(id_string=asset.uid)
815             parser = XFormInstanceParser(xform.xml, xform.data_dictionary())
816  
817             # Access the first child element of the <instance> node
818             instance_node = parser.get_root_node().getElementsByTagName('instance')[0]
819             root_element = instance_node.firstChild
(Pdb) import json
(Pdb) print(json.dumps(asset.content, indent=2))
{
  "survey": [],
  "settings": {
    "name": "custom_root_node_name"
  },
  "translations": [
    null
  ],
  "translated": [
    "label"
  ],
  "schema": "1"
}
```

Reply:
> Interesting! I'm surprised the unit tests don't fail since there's no `type`. I'll look into it.

I'm ashamed. I completely misunderstood this and replaced `type` with `text` 🤦
Thanks @jnm  for the correction :-)

Reply:
Made an internal thread to discuss the weird (existing) behavior of silently discarding invalid rows:
https://chat.kobotoolbox.org/#narrow/stream/4-Kobo-Dev/topic/Invalid.20asset.20content.20rows.20silently.20disarded